engine/runtime/src/main/scala/org/enso/compiler/phase/SymbolsImportResolution.scala

```diff
@@ -0,0 +1,562 @@
+      Array[Object](symbolToImport, typeName, module.getName)
+    )
+    val bindingMap = module.getIr.unsafeGetMetadata(
+      BindingAnalysis, "Should be here"
+    )
+    if (typeName.isDefined) {
+      bindingMap.exportedSymbols.get(typeName.get) match {
+        case Some(foundNamesForTypeName) =>
+          foundNamesForTypeName.collectFirst { case x: BindingsMap.ResolvedType => x } match {
+            case Some(resolvedType) =>
+              // Find `symbolToImport` member from `resolvedType`
+              val foundConstructor = resolvedType.tp.members.find(_.name == symbolToImport.name)
+              foundConstructor match {
+                case Some(ctor) => Left(BindingsMap.ResolvedConstructor(resolvedType, ctor))
+                case None =>
+                  Right(
+                    IR.Error.ImportExport.NoSuchConstructor(
+                      resolvedType.tp.name,
+                      symbolToImport.name
+                    )
+                  )
+              }
+            case None =>
+              Right(
+                IR.Error.ImportExport.TypeDoesNotExist(
+                  symbolToImport.name,
+                  module.getName.toString
+                )
+              )
+          }
+        case None =>
+          Right(
+            IR.Error.ImportExport.TypeDoesNotExist(
+              symbolToImport.name,
+              module.getName.toString
+            )
+          )
+      }
+    } else {
+      bindingMap.exportedSymbols.get(symbolToImport.name) match {
+        case Some(resolvedNames) =>
+          val resolvedImportTargets: List[ImportTarget] =
+            resolvedNames.collect {
+              case resolvedModule: BindingsMap.ResolvedModule => resolvedModule
+              case resolvedType: BindingsMap.ResolvedType => resolvedType
+              case resolvedMethod: BindingsMap.ResolvedMethod => resolvedMethod
+              case resolvedCtor: BindingsMap.ResolvedConstructor => resolvedCtor
+            }
+          // Take the first resolvedName that is also ImportTarget
+          resolvedImportTargets.headOption match {
+            case Some(resolvedImportTarget) => Left(resolvedImportTarget)
+            case None => Right(
+              IR.Error.ImportExport.SymbolsDoNotExist(
+                List(symbolToImport.name),
+                module.getName.toString
+              )
+            )
+          }
+        case None =>
+          Right(
+            IR.Error.ImportExport.SymbolsDoNotExist(
+              List(symbolToImport.name),
+              module.getName.toString
+            )
+          )
+      }
+    }
+  }
+
+  private def typeExists(
+    typeName: String,
+    importedBindingMap: BindingsMap
+  ): Boolean = {
+    importedBindingMap.definedEntities.exists(entity => {
+      entity match {
+        case BindingsMap.Type(name, _, _) if name == typeName => true
+        case _ => false
+      }
+    })
+  }
+
+  private case class SymbolsResolution(
```

Comment:
`Either[IR.Error.ImportExport.Reason, List[ImportTarget]]` seems, indeed, as a better type for the symbol resolution result. Originally, I added `SymbolsResolution` to be able to collect both errors and some resolved symbols - so that we have at least a partial result. But partial result is not a good idea. It should, indeed, be either an error or a successful list of resolved symbols.